distro/common/knot.service

```diff
@@ -10,6 +10,49 @@ User=knot
 Group=knot
 CapabilityBoundingSet=CAP_NET_BIND_SERVICE CAP_SETPCAP
 AmbientCapabilities=CAP_NET_BIND_SERVICE CAP_SETPCAP
+# Require systemd >= v238
+#TemporaryFileSystem=/run:ro /var:ro
+# Require systemd >= v233
+#BindPaths=/run/systemd
+#BindPaths=/run/knot
+#BindPaths=/var/lib/knot
+# Require systemd >= v235
+#LockPersonality=yes
+# Require systemd >= v231
+#MemoryDenyWriteExecute=yes
+# Require systemd >= v239
+#NoNewPrivileges=yes
+PrivateDevices=yes
+PrivateTmp=yes
+# Require systemd >= v232
+#ProtectControlGroups=yes
+ProtectHome=yes
+# Require systemd >= v242
+#ProtectHostname=yes
+# Require systemd >= v245
+#ProtectClock=yes
+# Require systemd >= v244
+#ProtectKernelLogs=yes
+# Require systemd >= v232
+#ProtectKernelModules=yes
+#ProtectKernelTunables=yes
+# Require systemd >= v232 (strict), v214
+# and ReadWritePaths= OR TemporaryFileSystem= AND BindPaths=
+#ProtectSystem=strict
+# Require systemd >= v232
+#RemoveIPC=yes
+RestrictAddressFamilies=AF_UNIX AF_INET AF_INET6
+# Require systemd >= v233
+#RestrictNamespaces=yes
+# Require systemd >= v231
+#RestrictRealtime=yes
+# Require systemd >= v242
+#RestrictSUIDSGID=yes
+SystemCallArchitectures=native
+SystemCallErrorNumber=EPERM
+# Require systemd >= v187, v239 (@system-service), v232(@resources)
+#SystemCallFilter=@system-service
```

Comment:
I don't think the benefits of `SystemCallFilter=` outweigh the potential issues.

I can already see how some new feature is added, which utilizes a different system call than is permitted in this filter. I don't think anyone runs knot under systemd during development, so this option likely won't be updated. Then the software will be released, packaged, and the new feature won't work (or worse, crash the daemon) for users, because of this option.

Reply:
If a whitelist is too risky, would a blacklist work? I think it is unlikely that knot will ever require; `@swap`, `@reboot`, `@cpu-emulation`, `@module` (I think that is already blacklisted due to `ProtectKernelModules=yes`), `@mount` or `@obsolete` syscalls. Maybe also `@debug` and `@raw-io`?

https://www.freedesktop.org/software/systemd/man/systemd.exec.html#SystemCallFilter=